src/settings.ts

```diff
@@ -0,0 +1,81 @@
+import * as vscode from "vscode";
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
+import { State } from "./extension";
+import { Project, Group, Filter, generateSvgUri } from './utils';
+
+const settingFilename: string = path.join(os.homedir(), 'vscode_log_analysis.json');
```

Comment:
store home dir can be messy to users --- the file exists after they uninstall the extension or vscode. 

I recommend context.globalStorageUri (context var from the activate function)

Reply:
That's good idea. thanks.

Reply:
Saw the change. I understand the need to store the Uri to some "global vairables" like state, but I don't think it is a good idea to pass state to "getSettingFile", "openSettings", "readSettings". It is better to just pass the Uri itself, which will make the state management less coupled.

Reply:
I fixed it.